homeassistant/components/ipma/__init__.py

```diff
@@ -1,19 +1,61 @@
 """Component for the Portuguese weather service - IPMA."""
+import logging
+
+import async_timeout
+from pyipma import IPMAException
+from pyipma.api import IPMA_API
+from pyipma.location import Location
+
 from homeassistant.config_entries import ConfigEntry
-from homeassistant.const import Platform
+from homeassistant.const import CONF_LATITUDE, CONF_LONGITUDE, Platform
 from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
 
 from .config_flow import IpmaFlowHandler  # noqa: F401
-from .const import DOMAIN  # noqa: F401
+from .const import DATA_API, DATA_LOCATION, DOMAIN
 
 DEFAULT_NAME = "ipma"
 
 PLATFORMS = [Platform.WEATHER]
 
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_get_api(hass):
```

Comment:
Why do we have a function to create the client? It only seems to be used once. We can just inline this.